src/gil.rs

```diff
@@ -224,19 +207,11 @@ impl GILGuard {
         let gstate = ffi::PyGILState_Ensure(); // acquire GIL
         increment_gil_count();
 
-        #[cfg(feature = "gil-refs")]
-        #[allow(deprecated)]
-        let pool = mem::ManuallyDrop::new(GILPool::new());
-
         #[cfg(not(pyo3_disable_reference_pool))]
         if let Some(pool) = Lazy::get(&POOL) {
             pool.update_counts(Python::assume_gil_acquired());
         }
```

Comment:
Can we get rid of this too?

Reply:
Sadly not, the "reference pool" is needed for delayed decrefs and so stays around, it's separate from the "GIL Pool" which the GIL Refs were stored in.